### PR TITLE
Adding unit test for getSauceBuildAction

### PR DESCRIPTION
**Pull request by moizjv**
@halkeye what do you think about this?

I am still figuring out how can we unit test this better

### CODE REVIEW

**Comment by halkeye** [comment]
You may want to try to rollback to an earlier version and see if this change catches the problem

**Comment by moizjv** [comment]
na this is just testing the newly added function, tried few things create a maven job that can catch earlier bug unsuccessfully will try few more tricks

**Comment by moizjv** [comment]
Ok i have added couple of test which tests the newly added `getSauceBuildAction` function, I think we need to add more coverage for tests, I would like that to be part of quality time.

File: src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java [hunk 1]
```diff
@@ -69,6 +74,46 @@ protected JenkinsSauceREST getSauceREST() {
...
+        SauceOnDemandBuildAction buildAction = new SauceOnDemandBuildAction(build, bw.getCredentialId());
+        build.addAction(buildAction);
+        SauceOnDemandBuildAction  sauceBuildAction= SauceOnDemandBuildAction.getSauceBuildAction(build);
+        assertEquals("fakeuser", sauceBuildAction.getCredentials().getUsername());
+    }
+
+    /**
+     * Verifies get sauce action for maven build
+     *
+     * @throws Exception
```
Comment (halkeye) [inline src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java]: this throws an exception?

File: src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java [hunk 2]
```diff
@@ -85,6 +89,30 @@ public void testGetSauceBuildAction () throws Exception {
...
+        );
+        project.getBuildWrappersList().add(bw);
+        project.setScm(new SingleFileSCM("pom.xml",getClass().getResource("/pom.xml")));
+        project.setGoals("clean");
+        MavenModuleSetBuild build =  project.scheduleBuild2(0).get(1, TimeUnit.MINUTES);
+        SauceOnDemandBuildAction buildAction = new SauceOnDemandBuildAction(build, bw.getCredentialId());
+        build.addAction(buildAction);
+        final MavenBuild mavenBuildMock = mock(MavenBuild.class);
+        when(mavenBuildMock.getParentBuild()).thenReturn(build);
+        SauceOnDemandBuildAction  sauceBuildAction= SauceOnDemandBuildAction.getSauceBuildAction(mavenBuildMock);
```
Comment (halkeye) [inline src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java]: nitpicky, but do you want to fix the spacing?
you have 2 before sauceBuildAction, and 0 after it

**Comment by halkeye** [comment]
I'm going to assume you removed the fix and tried it and it failed? 'cause i'm not 100% sure I understand that the maven test case you created makes sub projects.

File: src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java (hunk 1 above)
Comment (moizjv) [inline src/test/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildActionTest.java]: that is a copy past issue :/, will remove those comments.

**Comment by moizjv** [comment]
I am going to spend more time tmrw in quality time on this. Is this good to merge? Again I have opened a PR in upstream :( will close this and push to master

**Comment by halkeye** [comment]
yea its good to merge

You can also just merge it to upstream and then git pull upstream master into our fork.

I think thats a better idea 'cause it keeps some history

**Comment by moizjv** [comment]
cool :)

**Pull request merged**